coding/javascript.md

```diff
@@ -37,6 +37,26 @@ We use [Yarn](https://yarnpkg.com/en) for dependency management.
 
 We use the [Jest](https://jestjs.io) unittest framework.
 
+#### General Testing Principals
+  * Avoid setting up state in `beforeEach`. It should be clear from reading a test in isolation what state it is in.
```

Comment:
Could we define in a little more detail what is meant by "state" here?

Reply:
@nottrobin @bartaz have expanded on this a bit, hope it is clearer now.